enode.rkt

```diff
@@ -233,9 +236,20 @@
 (define (enode-vars en)
   (enode-cvars (pack-leader en)))
 
+(define (update-en-expr expr)
+  (if (list? expr)
+      (for/list ([sub (in-list expr)])
+        (if (enode? sub) (pack-leader sub) sub))
+      expr))
+
+
 ;; Removes duplicates from the varset of this node.
 (define (refresh-vars! en)
-  (set-enode-cvars! en (list->set (set->list (enode-cvars en)))))
+  (set-enode-cvars!
```

Comment:
**Bug 4**
cvars was never being updated when we merge eclasses. Now we canonicalize all of them and deduplicate